realm/realm-library/src/main/java/io/realm/internal/RealmNotifier.java

```diff
@@ -97,10 +98,16 @@ protected RealmNotifier(SharedRealm sharedRealm) {
     // called from java_binding_context.cpp
     void didChange() {
         realmObserverPairs.foreach(onChangeCallBack);
-        for (Runnable runnable : transactionCallbacks) {
-            runnable.run();
+
+        if (!transactionCallbacks.isEmpty()) {
+            // The callback list needs to be cleared before calling to avoid sync transactions in the callback
```

Comment:
`sync transactions` is a bit ambiguous since we have `sync` feature and it also uses transaction.

Does `synced transactions` or `synchronized transactions` solve this?